darts/utils/timeseries_generation.py

```diff
@@ -206,10 +206,42 @@ def random_walk_timeseries(length: int = 10,
     return TimeSeries.from_times_and_values(times, values, freq=freq)
 
 
+def _extend_time_index_until(time_index:pd.DatetimeIndex,
+                             until: Union[int, str, pd.Timestamp],
+                             ) -> pd.DatetimeIndex:
+    if until == 0: return time_index
+
+    end = time_index[-1]
+    freq = time_index.freq
```

Comment:
If instead you take the `freq` attribute of a `TimeSeries`, it will be `1` if the series is integer-indexed (i.e., indexed with a `pandas.Int64Index` instead of a `pandas.DatetimeIndex`). This can be convenient for doing arithmetic on timestamps in a way that is agnostic to whether they are ints or Timestamps (e.g. what you're doing at line 221 would also work on integer-indexed series).